### PR TITLE
Migrate to mcvine

### PR DESCRIPTION
**Pull request by ilumsden**
Migrates the Python code and tests in SCADGen to conform with McVine convention instead of pyre convention. The full "list" of updates made by this pull request can be found in #6 and #7.

### CODE REVIEW

**Comment by ilumsden** [comment]
Also, since this branch was created from the `travis` branch, some of the incomplete travis work would be added by this pull request. Hopefully I'll be able to finish the work on TravisCI relatively soon, so this shouldn't be a big problem besides it being annoying.

File: SCADGen/python/Parser.py [hunk 1]
```diff
@@ -91,7 +85,7 @@ def isComp(self, elem):
         returns False.
         """
         tag = elem.tag
-        if tag == "block" or tag == "cone" or tag == "cylinder" or tag == "sphere" or tag == "torus":
+        if tag == "block" or tag == "cone" or tag == "cylinder" or tag == "pyramid" or tag == "sphere" or tag == "torus":
```
Comment (granrothge) [inline SCADGen/python/Parser.py]: try changing to the 
``` if tag in ["block","cone","cylinder","pyramid","sphere","torus"] :  ``` syntax

**Comment by yxqd** [comment]
@ilumsden Thanks for the comments on the units. I added a check https://github.com/mcvine/SCADGen/blob/22f93c4e5e39188612ec297d1c16fcbb99cdb928/SCADGen/python/components/__init__.py#L7 and slightly refactored how the length unit conversion is done in the solids. See 22f93c4

**Comment by yxqd** [comment]
I will just merge this PR and create a new ticket on styling improvements.

**Pull request merged**